src/services/http-client.js

```diff
@@ -7,10 +8,29 @@ export const httpClient = async (url, parameters) =>
     });
 
 export function* APICall(endpoint, parameters) {
-    let token = yield select((state) => state.auth.accessToken);
+    let { accessToken, refreshToken, expiry } = yield select((state) => state.auth);
+
+    // check for 69 seconds before expiry :)
+    if (new Date() >= new Date((expiry - 30) * 1000)) {
+        try {
```

Comment:
This seems like this should have its own saga and actions. My personal opinion, could be kept like this as well but then redux logger wont show refresh token action. just a loginsuccess while seems confusing
![image](https://user-images.githubusercontent.com/43286669/111472459-fa833100-874f-11eb-85cc-e54dc197b552.png)

Reply:
I mean, the logger is going to be removed in production anyway, so console output shouldn't be an issue at all

Reply:
Also, won't making it a separate saga mean that we'll have to dispatch the action, and then wait till we get a refresh success to make the original API call which sounds unnecessarily complicated. Unless you want to call the saga function directly, which also seems to be a little weird cuz then why would you make it a saga instead of a normal function.

Reply:
not exactly. You will need one action and saga code is your try catch